{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "import plotly.graph_objects as go\n",
    "import plotly.express as px\n",
    "import numpy as np"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Source of info\n",
    "\n",
    "- **LQ_gtdbtk.bac120.summary.tsv** - taxonomic info about representitives MAGs from LQ clusterin for Bacteria sp. Created with GTDB-tk v2.0 (1,641 lines)\n",
    "- **LQ_gtdbtk.ar53.summary.tsv** - taxonomic info about representitives MAGs from LQ clusterin for Archaea sp. Created with GTDB-tk v2.0 (5 lines)\n",
    "- **LQ_genomeInformation.csv** - info about completeness,contamination,strain_heterogeneity,length,N50. Created with dREP (15,430 lines)\n",
    "- **LQ_Cdb.csv** - info about secondary_cluster. Created with dREP (9,475 lines, only bins goin to clustering)\n",
    "- **2023_Pantiukh_EstMB_MAGsDB_Hqcl.csv** - list of representatives genomes from cluster. Created with 'ls * all dereplicated \n",
    "  \n",
    "### Ð¡alculated parameters \n",
    "- **bins_in_cluster** - calculated here based on final table\n",
    "- **bin_qual** - calculated here based on final table\n",
    "\n",
    "### Manualy added info\n",
    "- **rep_MAG_ID** - manually added. Format: [L][0001][sp_name] were H - LQ clustering, 0001 - number, based on bacteria prevalence, sp_name - in case of unknow undS (unknown species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>VZR2LZ_vamb.278282.fa</td>\n",
       "      <td>67.24</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>899993</td>\n",
       "      <td>41034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>VZSPJW_maxbin.036.fa</td>\n",
       "      <td>93.97</td>\n",
       "      <td>47.08</td>\n",
       "      <td>26.23</td>\n",
       "      <td>3167676</td>\n",
       "      <td>2880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>VZTMOW_vamb.137436.fa</td>\n",
       "      <td>15.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>65154</td>\n",
       "      <td>65154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>VZTMOW_vamb.164616.fa</td>\n",
       "      <td>67.63</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1364300</td>\n",
       "      <td>30903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>VZWEWM_maxbin.048.fa</td>\n",
       "      <td>33.62</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2294493</td>\n",
       "      <td>5647</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "15424  VZR2LZ_vamb.278282.fa         67.24           0.00   \n",
       "15425   VZSPJW_maxbin.036.fa         93.97          47.08   \n",
       "15426  VZTMOW_vamb.137436.fa         15.52           0.00   \n",
       "15427  VZTMOW_vamb.164616.fa         67.63           0.00   \n",
       "15428   VZWEWM_maxbin.048.fa         33.62           0.47   \n",
       "\n",
       "       strain_heterogeneity   length    N50  \n",
       "15424                  0.00   899993  41034  \n",
       "15425                 26.23  3167676   2880  \n",
       "15426                  0.00    65154  65154  \n",
       "15427                  0.00  1364300  30903  \n",
       "15428                  0.00  2294493   5647  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 0. Read genomeInformation file\n",
    "genomeInformation = pd.read_csv('input/LQ_genomeInformation.csv')\n",
    "# rename column genome\n",
    "genomeInformation = genomeInformation.rename(columns={'genome': 'bin_name'})\n",
    "# drop column centrality\n",
    "genomeInformation = genomeInformation.drop(columns=['centrality'])\n",
    "genomeInformation.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>VZR2LZ_vamb.278282.fa</td>\n",
       "      <td>67.24</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>899993</td>\n",
       "      <td>41034</td>\n",
       "      <td>LQ-225_1.2.5.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>VZSPJW_maxbin.036.fa</td>\n",
       "      <td>93.97</td>\n",
       "      <td>47.08</td>\n",
       "      <td>26.23</td>\n",
       "      <td>3167676</td>\n",
       "      <td>2880</td>\n",
       "      <td>LQ-326_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>VZTMOW_vamb.137436.fa</td>\n",
       "      <td>15.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>65154</td>\n",
       "      <td>65154</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>VZTMOW_vamb.164616.fa</td>\n",
       "      <td>67.63</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1364300</td>\n",
       "      <td>30903</td>\n",
       "      <td>LQ-738_1.2.3.4.5.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>VZWEWM_maxbin.048.fa</td>\n",
       "      <td>33.62</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2294493</td>\n",
       "      <td>5647</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "15424  VZR2LZ_vamb.278282.fa         67.24           0.00   \n",
       "15425   VZSPJW_maxbin.036.fa         93.97          47.08   \n",
       "15426  VZTMOW_vamb.137436.fa         15.52           0.00   \n",
       "15427  VZTMOW_vamb.164616.fa         67.63           0.00   \n",
       "15428   VZWEWM_maxbin.048.fa         33.62           0.47   \n",
       "\n",
       "       strain_heterogeneity   length    N50             cluster  \n",
       "15424                  0.00   899993  41034      LQ-225_1.2.5.6  \n",
       "15425                 26.23  3167676   2880            LQ-326_1  \n",
       "15426                  0.00    65154  65154           undefined  \n",
       "15427                  0.00  1364300  30903  LQ-738_1.2.3.4.5.9  \n",
       "15428                  0.00  2294493   5647           undefined  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 1. Add info about cluster\n",
    "cluster = pd.read_csv('input/LQ_Cdb.csv')\n",
    "# rename column genome\n",
    "cluster = cluster.rename(columns={'genome': 'bin_name'})\n",
    "# keep only bin name and cluster columns\n",
    "cluster = cluster[['bin_name', 'secondary_cluster']]\n",
    "cluster = cluster.rename(columns={'secondary_cluster': 'cluster'})\n",
    "# merge cluster info with genome info\n",
    "mrg_st1 = pd.merge(genomeInformation, cluster, on='bin_name', how='outer')\n",
    "# fill NaN values with value\n",
    "mrg_st1 = mrg_st1.fillna('undefined')\n",
    "# add prefix to cluster name if it is not undefined\n",
    "mrg_st1['cluster'] = 'LQ-'+mrg_st1['cluster'].astype(str)\n",
    "mrg_st1['cluster'] = mrg_st1['cluster'].replace('LQ-undefined', 'undefined')\n",
    "\n",
    "mrg_st1.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "      <td>68.97</td>\n",
       "      <td>25.86</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2406469</td>\n",
       "      <td>5865</td>\n",
       "      <td>LQ-457_24</td>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "      <td>86.80</td>\n",
       "      <td>26.88</td>\n",
       "      <td>44.00</td>\n",
       "      <td>3157159</td>\n",
       "      <td>5203</td>\n",
       "      <td>LQ-265_1</td>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "      <td>72.65</td>\n",
       "      <td>41.38</td>\n",
       "      <td>34.29</td>\n",
       "      <td>2934952</td>\n",
       "      <td>10499</td>\n",
       "      <td>LQ-793_2</td>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "      <td>65.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1468926</td>\n",
       "      <td>120694</td>\n",
       "      <td>LQ-696_1</td>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>49.29</td>\n",
       "      <td>74.00</td>\n",
       "      <td>3007490</td>\n",
       "      <td>1894</td>\n",
       "      <td>LQ-457_3</td>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "15424  VY25ST_vamb.218068.fa         68.97          25.86   \n",
       "15425    VY36ZZ_metabat.1.fa         86.80          26.88   \n",
       "15426   VYHIBD_maxbin.133.fa         72.65          41.38   \n",
       "15427  VYWBYK_metabat.123.fa         65.52           0.00   \n",
       "15428   VZGZLV_maxbin.027.fa        100.00          49.29   \n",
       "\n",
       "       strain_heterogeneity   length     N50    cluster           rep_MAG_name  \n",
       "15424                100.00  2406469    5865  LQ-457_24  VY25ST_vamb.218068.fa  \n",
       "15425                 44.00  3157159    5203   LQ-265_1    VY36ZZ_metabat.1.fa  \n",
       "15426                 34.29  2934952   10499   LQ-793_2   VYHIBD_maxbin.133.fa  \n",
       "15427                  0.00  1468926  120694   LQ-696_1  VYWBYK_metabat.123.fa  \n",
       "15428                 74.00  3007490    1894   LQ-457_3   VZGZLV_maxbin.027.fa  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 2. Add info about representative genomes\n",
    "representative = pd.read_csv('input/2023_Pantiukh_EstMB_MAGsDB_LQcl.csv')\n",
    "# define cluster name for each representative MAGs \n",
    "mrg = pd.merge(representative, mrg_st1, left_on='rep_MAG_name', right_on='bin_name', how='inner')\n",
    "mrg = mrg[['rep_MAG_name', 'cluster']]\n",
    "# add line to mrg\n",
    "mrg.loc[len(mrg)] = ['undefined', 'undefined']\n",
    "# add ino about rep_MAG to mrg1\n",
    "mrg_st2 = pd.merge(mrg_st1, mrg, on='cluster', how='outer')\n",
    "\n",
    "mrg_st2.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "      <td>68.97</td>\n",
       "      <td>25.86</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2406469</td>\n",
       "      <td>5865</td>\n",
       "      <td>LQ-457_24</td>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales_A</td>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>Haemophilus_D</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "      <td>86.80</td>\n",
       "      <td>26.88</td>\n",
       "      <td>44.00</td>\n",
       "      <td>3157159</td>\n",
       "      <td>5203</td>\n",
       "      <td>LQ-265_1</td>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Lachnospirales</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>Coprococcus_A</td>\n",
       "      <td>Coprococcus_A catus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "      <td>72.65</td>\n",
       "      <td>41.38</td>\n",
       "      <td>34.29</td>\n",
       "      <td>2934952</td>\n",
       "      <td>10499</td>\n",
       "      <td>LQ-793_2</td>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Limivicinus</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "      <td>65.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1468926</td>\n",
       "      <td>120694</td>\n",
       "      <td>LQ-696_1</td>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Acutalibacteraceae</td>\n",
       "      <td>UMGS1858</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>49.29</td>\n",
       "      <td>74.00</td>\n",
       "      <td>3007490</td>\n",
       "      <td>1894</td>\n",
       "      <td>LQ-457_3</td>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales_A</td>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>Haemophilus_D</td>\n",
       "      <td>Haemophilus_D parainfluenzae_K</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "15424  VY25ST_vamb.218068.fa         68.97          25.86   \n",
       "15425    VY36ZZ_metabat.1.fa         86.80          26.88   \n",
       "15426   VYHIBD_maxbin.133.fa         72.65          41.38   \n",
       "15427  VYWBYK_metabat.123.fa         65.52           0.00   \n",
       "15428   VZGZLV_maxbin.027.fa        100.00          49.29   \n",
       "\n",
       "       strain_heterogeneity   length     N50    cluster  \\\n",
       "15424                100.00  2406469    5865  LQ-457_24   \n",
       "15425                 44.00  3157159    5203   LQ-265_1   \n",
       "15426                 34.29  2934952   10499   LQ-793_2   \n",
       "15427                  0.00  1468926  120694   LQ-696_1   \n",
       "15428                 74.00  3007490    1894   LQ-457_3   \n",
       "\n",
       "                rep_MAG_name    domain          phylum                class  \\\n",
       "15424  VY25ST_vamb.218068.fa  Bacteria  Pseudomonadota  Gammaproteobacteria   \n",
       "15425    VY36ZZ_metabat.1.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15426   VYHIBD_maxbin.133.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15427  VYWBYK_metabat.123.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15428   VZGZLV_maxbin.027.fa  Bacteria  Pseudomonadota  Gammaproteobacteria   \n",
       "\n",
       "                    order              family          genus  \\\n",
       "15424  Enterobacterales_A     Pasteurellaceae  Haemophilus_D   \n",
       "15425      Lachnospirales     Lachnospiraceae  Coprococcus_A   \n",
       "15426     Oscillospirales    Oscillospiraceae    Limivicinus   \n",
       "15427     Oscillospirales  Acutalibacteraceae       UMGS1858   \n",
       "15428  Enterobacterales_A     Pasteurellaceae  Haemophilus_D   \n",
       "\n",
       "                                   sp  \n",
       "15424                       undefined  \n",
       "15425             Coprococcus_A catus  \n",
       "15426                       undefined  \n",
       "15427                       undefined  \n",
       "15428  Haemophilus_D parainfluenzae_K  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 3. Add info about taxonomy\n",
    "\n",
    "# READ bacteria taxonomy\n",
    "gt = pd.read_csv('input/LQ_gtdbtk.bac120.summary.tsv', sep='\\t')\n",
    "gt = gt[['user_genome','classification','fastani_ani','closest_placement_reference','closest_placement_ani']]\n",
    "\n",
    "gt['domain'] = gt['classification'].str.split(';', expand=True)[0].str.replace(r'd__', '')\n",
    "gt['phylum'] = gt['classification'].str.split(';', expand=True)[1].str.replace(r'p__', '')\n",
    "gt['class'] = gt['classification'].str.split(';', expand=True)[2].str.replace(r'c__', '')\n",
    "gt['order'] = gt['classification'].str.split(';', expand=True)[3].str.replace(r'o__', '')\n",
    "gt['family'] = gt['classification'].str.split(';', expand=True)[4].str.replace(r'f__', '')\n",
    "gt['genus'] = gt['classification'].str.split(';', expand=True)[5].str.replace(r'g__', '')\n",
    "gt['sp'] = gt['classification'].str.split(';', expand=True)[6].str.replace(r's__', '')\n",
    "gt = gt.drop('classification', axis=1)\n",
    "\n",
    "# rename column genome\n",
    "gt = gt.rename(columns={'user_genome': 'rep_MAG_name'})\n",
    "# skip columns \n",
    "gt = gt[['rep_MAG_name', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'sp']]\n",
    "# fill NaN empty values with value\n",
    "gt['sp'] = gt['sp'].replace('', 'undefined')\n",
    "gt['genus'] = gt['genus'].replace('', 'undefined')\n",
    "gt['family'] = gt['family'].replace('', 'undefined')\n",
    "\n",
    "# READ archaea taxonomy\n",
    "ar = pd.read_csv('input/LQ_gtdbtk.ar53.summary.tsv', sep='\\t')\n",
    "ar = ar[['user_genome','classification','fastani_ani','closest_placement_reference','closest_placement_ani']]\n",
    "\n",
    "ar['domain'] = ar['classification'].str.split(';', expand=True)[0].str.replace(r'd__', '')\n",
    "ar['phylum'] = ar['classification'].str.split(';', expand=True)[1].str.replace(r'p__', '')\n",
    "ar['class'] = ar['classification'].str.split(';', expand=True)[2].str.replace(r'c__', '')\n",
    "ar['order'] = ar['classification'].str.split(';', expand=True)[3].str.replace(r'o__', '')\n",
    "ar['family'] = ar['classification'].str.split(';', expand=True)[4].str.replace(r'f__', '')\n",
    "ar['genus'] = ar['classification'].str.split(';', expand=True)[5].str.replace(r'g__', '')\n",
    "ar['sp'] = ar['classification'].str.split(';', expand=True)[6].str.replace(r's__', '')\n",
    "ar = ar.drop('classification', axis=1)\n",
    "\n",
    "# rename column genome\n",
    "ar = ar.rename(columns={'user_genome': 'rep_MAG_name'})\n",
    "# skip columns \n",
    "ar = ar[['rep_MAG_name', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'sp']]\n",
    "# fill NaN empty values with value\n",
    "ar['sp'] = ar['sp'].replace('', 'undefined')\n",
    "ar['genus'] = ar['genus'].replace('', 'undefined')\n",
    "\n",
    "# concat bacteria and archaea taxonomy\n",
    "taxa = pd.concat([gt, ar], ignore_index=True)\n",
    "# add fa to rep_MAG_name\n",
    "taxa['rep_MAG_name'] = taxa['rep_MAG_name'].astype(str)+'.fa'\n",
    "# add line to mrg\n",
    "taxa.loc[len(taxa)] = ['undefined', 'undefined','undefined', 'undefined','undefined', 'undefined','undefined', 'undefined']\n",
    "\n",
    "# add ino about taxonomy to mrg_st2\n",
    "mrg_st3 = pd.merge(mrg_st2, taxa, on='rep_MAG_name', how='outer')\n",
    "\n",
    "mrg_st3.tail()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15424</th>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "      <td>68.97</td>\n",
       "      <td>25.86</td>\n",
       "      <td>100.00</td>\n",
       "      <td>2406469</td>\n",
       "      <td>5865</td>\n",
       "      <td>LQ-457_24</td>\n",
       "      <td>VY25ST_vamb.218068.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales_A</td>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>Haemophilus_D</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15425</th>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "      <td>86.80</td>\n",
       "      <td>26.88</td>\n",
       "      <td>44.00</td>\n",
       "      <td>3157159</td>\n",
       "      <td>5203</td>\n",
       "      <td>LQ-265_1</td>\n",
       "      <td>VY36ZZ_metabat.1.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Lachnospirales</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>Coprococcus_A</td>\n",
       "      <td>Coprococcus_A catus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15426</th>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "      <td>72.65</td>\n",
       "      <td>41.38</td>\n",
       "      <td>34.29</td>\n",
       "      <td>2934952</td>\n",
       "      <td>10499</td>\n",
       "      <td>LQ-793_2</td>\n",
       "      <td>VYHIBD_maxbin.133.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Limivicinus</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15427</th>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "      <td>65.52</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1468926</td>\n",
       "      <td>120694</td>\n",
       "      <td>LQ-696_1</td>\n",
       "      <td>VYWBYK_metabat.123.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Acutalibacteraceae</td>\n",
       "      <td>UMGS1858</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15428</th>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>49.29</td>\n",
       "      <td>74.00</td>\n",
       "      <td>3007490</td>\n",
       "      <td>1894</td>\n",
       "      <td>LQ-457_3</td>\n",
       "      <td>VZGZLV_maxbin.027.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Enterobacterales_A</td>\n",
       "      <td>Pasteurellaceae</td>\n",
       "      <td>Haemophilus_D</td>\n",
       "      <td>Haemophilus_D parainfluenzae_K</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "15424  VY25ST_vamb.218068.fa         68.97          25.86   \n",
       "15425    VY36ZZ_metabat.1.fa         86.80          26.88   \n",
       "15426   VYHIBD_maxbin.133.fa         72.65          41.38   \n",
       "15427  VYWBYK_metabat.123.fa         65.52           0.00   \n",
       "15428   VZGZLV_maxbin.027.fa        100.00          49.29   \n",
       "\n",
       "       strain_heterogeneity   length     N50    cluster  \\\n",
       "15424                100.00  2406469    5865  LQ-457_24   \n",
       "15425                 44.00  3157159    5203   LQ-265_1   \n",
       "15426                 34.29  2934952   10499   LQ-793_2   \n",
       "15427                  0.00  1468926  120694   LQ-696_1   \n",
       "15428                 74.00  3007490    1894   LQ-457_3   \n",
       "\n",
       "                rep_MAG_name    domain          phylum                class  \\\n",
       "15424  VY25ST_vamb.218068.fa  Bacteria  Pseudomonadota  Gammaproteobacteria   \n",
       "15425    VY36ZZ_metabat.1.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15426   VYHIBD_maxbin.133.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15427  VYWBYK_metabat.123.fa  Bacteria     Bacillota_A           Clostridia   \n",
       "15428   VZGZLV_maxbin.027.fa  Bacteria  Pseudomonadota  Gammaproteobacteria   \n",
       "\n",
       "                    order              family          genus  \\\n",
       "15424  Enterobacterales_A     Pasteurellaceae  Haemophilus_D   \n",
       "15425      Lachnospirales     Lachnospiraceae  Coprococcus_A   \n",
       "15426     Oscillospirales    Oscillospiraceae    Limivicinus   \n",
       "15427     Oscillospirales  Acutalibacteraceae       UMGS1858   \n",
       "15428  Enterobacterales_A     Pasteurellaceae  Haemophilus_D   \n",
       "\n",
       "                                   sp  bin_number_in_cluster  \n",
       "15424                       undefined                      1  \n",
       "15425             Coprococcus_A catus                      1  \n",
       "15426                       undefined                      1  \n",
       "15427                       undefined                      1  \n",
       "15428  Haemophilus_D parainfluenzae_K                      1  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 4. Add info about bins number in cluster\n",
    "bn = mrg_st3[['bin_name','cluster']].groupby('cluster').count()\n",
    "# rename column\n",
    "bn = bn.rename(columns={'bin_name': 'bin_number_in_cluster'})\n",
    "# merge it with mrg_st3\n",
    "mrg_st4 = pd.merge(mrg_st3, bn, on='cluster', how='outer')\n",
    "\n",
    "mrg_st4.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SAVE FINAL TABLE with all bins went to LQcl\n",
    "mrg_st4.to_excel('results/LQcl_all_bins_all_info.xlsx', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1639</th>\n",
       "      <td>VZTMOW_maxbin.050_sub.fa</td>\n",
       "      <td>52.27</td>\n",
       "      <td>7.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1164295</td>\n",
       "      <td>1203</td>\n",
       "      <td>LQ-992_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobiota</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Opitutales</td>\n",
       "      <td>CAG-312</td>\n",
       "      <td>Merdousia</td>\n",
       "      <td>undefined</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>VZTMOW_maxbin.071.fa</td>\n",
       "      <td>66.77</td>\n",
       "      <td>25.86</td>\n",
       "      <td>41.18</td>\n",
       "      <td>1729642</td>\n",
       "      <td>49503</td>\n",
       "      <td>LQ-752_3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Vescimonas</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>VZTMOW_maxbin.072.fa</td>\n",
       "      <td>73.28</td>\n",
       "      <td>26.49</td>\n",
       "      <td>62.86</td>\n",
       "      <td>1821208</td>\n",
       "      <td>48328</td>\n",
       "      <td>LQ-752_2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Vescimonas</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>VZXIHP_vamb.46239.fa</td>\n",
       "      <td>73.75</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2515878</td>\n",
       "      <td>11062</td>\n",
       "      <td>LQ-683_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>UBA4372</td>\n",
       "      <td>UBA4372 sp900543815</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>VZZW6V_vamb.255038.fa</td>\n",
       "      <td>68.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>966784</td>\n",
       "      <td>22255</td>\n",
       "      <td>LQ-25_3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>TANB77</td>\n",
       "      <td>UBA1234</td>\n",
       "      <td>HGM13618</td>\n",
       "      <td>HGM13618 sp900753975</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  rep_MAG_name  completeness  contamination  \\\n",
       "1639  VZTMOW_maxbin.050_sub.fa         52.27           7.97   \n",
       "1640      VZTMOW_maxbin.071.fa         66.77          25.86   \n",
       "1641      VZTMOW_maxbin.072.fa         73.28          26.49   \n",
       "1642      VZXIHP_vamb.46239.fa         73.75           0.00   \n",
       "1643     VZZW6V_vamb.255038.fa         68.97           0.00   \n",
       "\n",
       "      strain_heterogeneity   length    N50   cluster    domain  \\\n",
       "1639                  0.00  1164295   1203  LQ-992_1  Bacteria   \n",
       "1640                 41.18  1729642  49503  LQ-752_3  Bacteria   \n",
       "1641                 62.86  1821208  48328  LQ-752_2  Bacteria   \n",
       "1642                  0.00  2515878  11062  LQ-683_1  Bacteria   \n",
       "1643                  0.00   966784  22255   LQ-25_3  Bacteria   \n",
       "\n",
       "                 phylum             class            order            family  \\\n",
       "1639  Verrucomicrobiota  Verrucomicrobiae       Opitutales           CAG-312   \n",
       "1640        Bacillota_A        Clostridia  Oscillospirales  Oscillospiraceae   \n",
       "1641        Bacillota_A        Clostridia  Oscillospirales  Oscillospiraceae   \n",
       "1642       Bacteroidota       Bacteroidia    Bacteroidales    Bacteroidaceae   \n",
       "1643        Bacillota_A        Clostridia           TANB77           UBA1234   \n",
       "\n",
       "           genus                    sp  bin_number_in_cluster  \n",
       "1639   Merdousia             undefined                      4  \n",
       "1640  Vescimonas             undefined                      1  \n",
       "1641  Vescimonas             undefined                      1  \n",
       "1642     UBA4372   UBA4372 sp900543815                      1  \n",
       "1643    HGM13618  HGM13618 sp900753975                      3  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 5. Select only representative MAGs\n",
    "mrg_st5 = pd.merge(representative, mrg_st4, left_on='rep_MAG_name', right_on='bin_name', how='inner')\n",
    "# rename column genome\n",
    "mrg_st5 = mrg_st5.rename(columns={'rep_MAG_name_x': 'rep_MAG_name'})\n",
    "# drop column centrality\n",
    "mrg_st5 = mrg_st5.drop(columns=['bin_name','rep_MAG_name_y'])\n",
    "mrg_st5.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>cluster</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "      <th>rep_MAG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1639</th>\n",
       "      <td>VZTMOW_maxbin.050_sub.fa</td>\n",
       "      <td>52.27</td>\n",
       "      <td>7.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1164295</td>\n",
       "      <td>1203</td>\n",
       "      <td>LQ-992_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Verrucomicrobiota</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Opitutales</td>\n",
       "      <td>CAG-312</td>\n",
       "      <td>Merdousia</td>\n",
       "      <td>undefined</td>\n",
       "      <td>4</td>\n",
       "      <td>L0475_Merdousia_undS.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1640</th>\n",
       "      <td>VZTMOW_maxbin.071.fa</td>\n",
       "      <td>66.77</td>\n",
       "      <td>25.86</td>\n",
       "      <td>41.18</td>\n",
       "      <td>1729642</td>\n",
       "      <td>49503</td>\n",
       "      <td>LQ-752_3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Vescimonas</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "      <td>L1642_Vescimonas_undS.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1641</th>\n",
       "      <td>VZTMOW_maxbin.072.fa</td>\n",
       "      <td>73.28</td>\n",
       "      <td>26.49</td>\n",
       "      <td>62.86</td>\n",
       "      <td>1821208</td>\n",
       "      <td>48328</td>\n",
       "      <td>LQ-752_2</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Oscillospiraceae</td>\n",
       "      <td>Vescimonas</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "      <td>L1643_Vescimonas_undS.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1642</th>\n",
       "      <td>VZXIHP_vamb.46239.fa</td>\n",
       "      <td>73.75</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2515878</td>\n",
       "      <td>11062</td>\n",
       "      <td>LQ-683_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>UBA4372</td>\n",
       "      <td>UBA4372 sp900543815</td>\n",
       "      <td>1</td>\n",
       "      <td>L1644_UBA4372_sp900543815.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1643</th>\n",
       "      <td>VZZW6V_vamb.255038.fa</td>\n",
       "      <td>68.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>966784</td>\n",
       "      <td>22255</td>\n",
       "      <td>LQ-25_3</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>TANB77</td>\n",
       "      <td>UBA1234</td>\n",
       "      <td>HGM13618</td>\n",
       "      <td>HGM13618 sp900753975</td>\n",
       "      <td>3</td>\n",
       "      <td>L0578_HGM13618_sp900753975.fa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  rep_MAG_name  completeness  contamination  \\\n",
       "1639  VZTMOW_maxbin.050_sub.fa         52.27           7.97   \n",
       "1640      VZTMOW_maxbin.071.fa         66.77          25.86   \n",
       "1641      VZTMOW_maxbin.072.fa         73.28          26.49   \n",
       "1642      VZXIHP_vamb.46239.fa         73.75           0.00   \n",
       "1643     VZZW6V_vamb.255038.fa         68.97           0.00   \n",
       "\n",
       "      strain_heterogeneity   length    N50   cluster    domain  \\\n",
       "1639                  0.00  1164295   1203  LQ-992_1  Bacteria   \n",
       "1640                 41.18  1729642  49503  LQ-752_3  Bacteria   \n",
       "1641                 62.86  1821208  48328  LQ-752_2  Bacteria   \n",
       "1642                  0.00  2515878  11062  LQ-683_1  Bacteria   \n",
       "1643                  0.00   966784  22255   LQ-25_3  Bacteria   \n",
       "\n",
       "                 phylum             class            order            family  \\\n",
       "1639  Verrucomicrobiota  Verrucomicrobiae       Opitutales           CAG-312   \n",
       "1640        Bacillota_A        Clostridia  Oscillospirales  Oscillospiraceae   \n",
       "1641        Bacillota_A        Clostridia  Oscillospirales  Oscillospiraceae   \n",
       "1642       Bacteroidota       Bacteroidia    Bacteroidales    Bacteroidaceae   \n",
       "1643        Bacillota_A        Clostridia           TANB77           UBA1234   \n",
       "\n",
       "           genus                    sp  bin_number_in_cluster  \\\n",
       "1639   Merdousia             undefined                      4   \n",
       "1640  Vescimonas             undefined                      1   \n",
       "1641  Vescimonas             undefined                      1   \n",
       "1642     UBA4372   UBA4372 sp900543815                      1   \n",
       "1643    HGM13618  HGM13618 sp900753975                      3   \n",
       "\n",
       "                         rep_MAG_ID  \n",
       "1639        L0475_Merdousia_undS.fa  \n",
       "1640       L1642_Vescimonas_undS.fa  \n",
       "1641       L1643_Vescimonas_undS.fa  \n",
       "1642   L1644_UBA4372_sp900543815.fa  \n",
       "1643  L0578_HGM13618_sp900753975.fa  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 6. Add info about rep_MAG_ID\n",
    "mid = pd.read_csv('input/LQ_rename.csv', sep='\\t')\n",
    "mrg_st6 = pd.merge(mrg_st5, mid, on='rep_MAG_name', how='inner')\n",
    "mrg_st6.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 7. Add info about bins quality\n",
    "conditions = [(mrg_st6['completeness'] > 90) & (mrg_st6['contamination'] < 5),\n",
    "        (mrg_st6['completeness'] <= 90) & (mrg_st6['contamination'] < 10) | \\\n",
    "        (mrg_st6['completeness'] > 90) & (mrg_st6['contamination'] >= 5) & (mrg_st6['contamination'] < 10),\n",
    "        (mrg_st6['contamination'] >= 10)]\n",
    "\n",
    "values = ['HQ', 'MQ', 'LQ']\n",
    "mrg_st6['bin_qual'] = np.select(conditions, values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SAVE FINAL TABLE with only representative MAGs came from LQcl\n",
    "mrg_st6.to_excel('results/LQcl_representative_MAGs_all_info.xlsx', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
