{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "import plotly.express as px\n",
    "import plotly.graph_objects as go\n",
    "import plotly.express as px\n",
    "import numpy as np"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Source of info\n",
    "\n",
    "- **HQ_gtdbtk.bac120.summary.tsv** - taxonomic info about representitives MAGs from LQ clusterin for Bacteria sp. Created with GTDB-tk v2.0 (2,256 lines)\n",
    "- **HQ_genomeInformation folder** - info about completeness,contamination,strain_heterogeneity,length,N50. Created with dREP (84,761 lines in 10 files. All bins including LQcl bins)\n",
    "- **HQ_Cdb folder** - info about pre-cluster. Created with dREP (69,332 lines)\n",
    "- **HQ_Cdb.csv** - info about final clusters. Created with dREP (9,815 lines)\n",
    "- **2023_Pantiukh_EstMB_MAGsDB_Hqcl.csv** - list of representatives genomes from cluster. Created with 'ls * all dereplicated \n",
    "  \n",
    "### Ð¡alculated parameters \n",
    "- **bins_in_cluster** - calculated here based on final table\n",
    "- **bin_qual** - calculated here based on final table\n",
    "\n",
    "### Manualy added info\n",
    "- **rep_MAG_ID** - manually added. Format: [H][0001][sp_name] were H - HQ clustering, 0001 - number, based on bacteria prevalence, sp_name - in case of unknow undS (unknown species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>84757</th>\n",
       "      <td>VZZW6V_vamb.182971.fa</td>\n",
       "      <td>87.93</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1489809</td>\n",
       "      <td>19724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84758</th>\n",
       "      <td>VZZW6V_vamb.205379.fa</td>\n",
       "      <td>81.03</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.0</td>\n",
       "      <td>983639</td>\n",
       "      <td>85642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84759</th>\n",
       "      <td>VZZW6V_vamb.213811.fa</td>\n",
       "      <td>34.48</td>\n",
       "      <td>5.17</td>\n",
       "      <td>100.0</td>\n",
       "      <td>877739</td>\n",
       "      <td>96766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84760</th>\n",
       "      <td>VZZW6V_vamb.51578.fa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1571046</td>\n",
       "      <td>525634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84761</th>\n",
       "      <td>VZZW6V_vamb.76515.fa</td>\n",
       "      <td>99.14</td>\n",
       "      <td>13.79</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1752869</td>\n",
       "      <td>228975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  completeness  contamination  \\\n",
       "84757  VZZW6V_vamb.182971.fa         87.93           0.00   \n",
       "84758  VZZW6V_vamb.205379.fa         81.03           1.72   \n",
       "84759  VZZW6V_vamb.213811.fa         34.48           5.17   \n",
       "84760   VZZW6V_vamb.51578.fa        100.00           0.00   \n",
       "84761   VZZW6V_vamb.76515.fa         99.14          13.79   \n",
       "\n",
       "       strain_heterogeneity   length     N50  \n",
       "84757                   0.0  1489809   19724  \n",
       "84758                   0.0   983639   85642  \n",
       "84759                 100.0   877739   96766  \n",
       "84760                   0.0  1571046  525634  \n",
       "84761                 100.0  1752869  228975  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 0. Read 10 genomeInformation files.\n",
    "# Read 10 genomeInformation files for pre-clustering process. Get all bins includes LQ bins (84,761 lines)\n",
    "genomeInformation = pd.DataFrame(columns=['genome', 'completeness', 'contamination', 'strain_heterogeneity','length', 'N50', 'centrality'])\n",
    "\n",
    "for n in range(1,11):\n",
    "\n",
    "    gen = pd.read_csv('./input/HQ_genomeInformation/genomeInformation_'+str(n)+'.csv')\n",
    "    genomeInformation = pd.concat([genomeInformation, gen], ignore_index=True)\n",
    "\n",
    "# rename column genome\n",
    "genomeInformation = genomeInformation.rename(columns={'genome': 'bin_name'})\n",
    "# drop column centrality\n",
    "genomeInformation = genomeInformation.drop(columns=['centrality'])  \n",
    "    \n",
    "genomeInformation.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69328</th>\n",
       "      <td>VZY957_maxbin.064_sub.fa</td>\n",
       "      <td>pre10-63_1</td>\n",
       "      <td>77.59</td>\n",
       "      <td>20.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4519669</td>\n",
       "      <td>2971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69329</th>\n",
       "      <td>VZY957_vamb.139249.fa</td>\n",
       "      <td>pre10-677_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2292055</td>\n",
       "      <td>54149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69330</th>\n",
       "      <td>VZY957_vamb.70285.fa</td>\n",
       "      <td>pre10-51_1</td>\n",
       "      <td>90.88</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1482390</td>\n",
       "      <td>23601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69331</th>\n",
       "      <td>VZZW6V_maxbin.057.fa</td>\n",
       "      <td>pre10-573_1</td>\n",
       "      <td>88.87</td>\n",
       "      <td>16.80</td>\n",
       "      <td>8.51</td>\n",
       "      <td>1895208</td>\n",
       "      <td>1552</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69332</th>\n",
       "      <td>VZZW6V_vamb.205379.fa</td>\n",
       "      <td>pre10-141_1</td>\n",
       "      <td>81.03</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>983639</td>\n",
       "      <td>85642</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       bin_name  pre_cluster  completeness  contamination  \\\n",
       "69328  VZY957_maxbin.064_sub.fa   pre10-63_1         77.59          20.14   \n",
       "69329     VZY957_vamb.139249.fa  pre10-677_1         91.38           0.00   \n",
       "69330      VZY957_vamb.70285.fa   pre10-51_1         90.88           0.00   \n",
       "69331      VZZW6V_maxbin.057.fa  pre10-573_1         88.87          16.80   \n",
       "69332     VZZW6V_vamb.205379.fa  pre10-141_1         81.03           1.72   \n",
       "\n",
       "       strain_heterogeneity   length    N50  \n",
       "69328                  0.00  4519669   2971  \n",
       "69329                  0.00  2292055  54149  \n",
       "69330                  0.00  1482390  23601  \n",
       "69331                  8.51  1895208   1552  \n",
       "69332                  0.00   983639  85642  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 1. Add info about pre-cluster (69,332 lines in 10 files)\n",
    "cluster = pd.DataFrame(columns=['bin_name', 'pre_cluster'])\n",
    "\n",
    "for n in range(1, 11):\n",
    "    clst = pd.read_csv('./input/HQ_Cdb/Cdb_' + str(n) + '.csv')\n",
    "    # rename column genome\n",
    "    clst = clst.rename(columns={'genome': 'bin_name'})\n",
    "    # keep only bin name and cluster columns\n",
    "    clst = clst[['bin_name', 'secondary_cluster']]\n",
    "    clst = clst.rename(columns={'secondary_cluster': 'pre_cluster'})\n",
    "    clst['pre_cluster'] = 'pre'+str(n)+'-'+clst['pre_cluster'].astype(str)\n",
    "    cluster = pd.concat([cluster, clst], ignore_index=True)\n",
    "\n",
    "# merge cluster info with genome info\n",
    "mrg_st1 = pd.merge(cluster, genomeInformation, on='bin_name', how='inner')\n",
    "mrg_st1.tail()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69328</th>\n",
       "      <td>VZY957_maxbin.064_sub.fa</td>\n",
       "      <td>pre10-63_1</td>\n",
       "      <td>77.59</td>\n",
       "      <td>20.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4519669</td>\n",
       "      <td>2971</td>\n",
       "      <td>VZY957_maxbin.064_sub.fa</td>\n",
       "      <td>HQ-227_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69329</th>\n",
       "      <td>VZY957_vamb.139249.fa</td>\n",
       "      <td>pre10-677_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2292055</td>\n",
       "      <td>54149</td>\n",
       "      <td>VZY957_vamb.139249.fa</td>\n",
       "      <td>HQ-645_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69330</th>\n",
       "      <td>VZY957_vamb.70285.fa</td>\n",
       "      <td>pre10-51_1</td>\n",
       "      <td>90.88</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1482390</td>\n",
       "      <td>23601</td>\n",
       "      <td>VZY957_vamb.70285.fa</td>\n",
       "      <td>HQ-230_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69331</th>\n",
       "      <td>VZZW6V_maxbin.057.fa</td>\n",
       "      <td>pre10-573_1</td>\n",
       "      <td>88.87</td>\n",
       "      <td>16.80</td>\n",
       "      <td>8.51</td>\n",
       "      <td>1895208</td>\n",
       "      <td>1552</td>\n",
       "      <td>VZZW6V_maxbin.057.fa</td>\n",
       "      <td>HQ-552_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69332</th>\n",
       "      <td>VZZW6V_vamb.205379.fa</td>\n",
       "      <td>pre10-141_1</td>\n",
       "      <td>81.03</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>983639</td>\n",
       "      <td>85642</td>\n",
       "      <td>VZZW6V_vamb.205379.fa</td>\n",
       "      <td>HQ-384_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       bin_name  pre_cluster  completeness  contamination  \\\n",
       "69328  VZY957_maxbin.064_sub.fa   pre10-63_1         77.59          20.14   \n",
       "69329     VZY957_vamb.139249.fa  pre10-677_1         91.38           0.00   \n",
       "69330      VZY957_vamb.70285.fa   pre10-51_1         90.88           0.00   \n",
       "69331      VZZW6V_maxbin.057.fa  pre10-573_1         88.87          16.80   \n",
       "69332     VZZW6V_vamb.205379.fa  pre10-141_1         81.03           1.72   \n",
       "\n",
       "       strain_heterogeneity   length    N50        bst_preCl_bin_name  \\\n",
       "69328                  0.00  4519669   2971  VZY957_maxbin.064_sub.fa   \n",
       "69329                  0.00  2292055  54149     VZY957_vamb.139249.fa   \n",
       "69330                  0.00  1482390  23601      VZY957_vamb.70285.fa   \n",
       "69331                  8.51  1895208   1552      VZZW6V_maxbin.057.fa   \n",
       "69332                  0.00   983639  85642     VZZW6V_vamb.205379.fa   \n",
       "\n",
       "        cluster  \n",
       "69328  HQ-227_1  \n",
       "69329  HQ-645_1  \n",
       "69330  HQ-230_1  \n",
       "69331  HQ-552_1  \n",
       "69332  HQ-384_1  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 1a. Add info about final HQ cluster (9,815 lines in file)\n",
    "fclst = pd.read_csv('./input/HQ_Cdb.csv')\n",
    "# rename column genome\n",
    "fclst = fclst.rename(columns={'genome': 'bst_preCl_bin_name'})\n",
    "fclst = fclst.rename(columns={'secondary_cluster': 'cluster'})\n",
    "\n",
    "fclst['cluster'] = 'HQ-'+fclst['cluster'].astype(str)\n",
    "fclst = fclst[['bst_preCl_bin_name', 'cluster']]\n",
    "mrg = pd.merge(fclst, mrg_st1, left_on='bst_preCl_bin_name', right_on='bin_name', how='inner')\n",
    "mrg = mrg[['bst_preCl_bin_name','pre_cluster','cluster']]\n",
    "# add info about final cluster to the merged table\n",
    "mrg_st1a = pd.merge(mrg_st1, mrg, on='pre_cluster', how='outer')\n",
    "#\n",
    "mrg_st1a.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69328</th>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>pre10-73_1</td>\n",
       "      <td>93.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2279068</td>\n",
       "      <td>46770</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>HQ-9_1</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69329</th>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>pre10-735_1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2448674</td>\n",
       "      <td>149658</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>HQ-52_1</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69330</th>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>pre10-305_1</td>\n",
       "      <td>92.24</td>\n",
       "      <td>2.59</td>\n",
       "      <td>33.33</td>\n",
       "      <td>2452983</td>\n",
       "      <td>20640</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>HQ-1099_1</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69331</th>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>pre10-338_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1774424</td>\n",
       "      <td>55805</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>HQ-1187_1</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69332</th>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>pre10-454_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>6.35</td>\n",
       "      <td>85.71</td>\n",
       "      <td>1595826</td>\n",
       "      <td>6604</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>HQ-1822_1</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  pre_cluster  completeness  contamination  \\\n",
       "69328   VZQLWU_vamb.11899.fa   pre10-73_1         93.97           0.00   \n",
       "69329   VZWEWM_maxbin.066.fa  pre10-735_1        100.00           1.72   \n",
       "69330  VZXIHP_metabat.131.fa  pre10-305_1         92.24           2.59   \n",
       "69331   VZXIHP_metabat.45.fa  pre10-338_1         98.28           1.72   \n",
       "69332   VZXIHP_metabat.71.fa  pre10-454_1         91.38           6.35   \n",
       "\n",
       "       strain_heterogeneity   length     N50     bst_preCl_bin_name  \\\n",
       "69328                  0.00  2279068   46770   VZQLWU_vamb.11899.fa   \n",
       "69329                  0.00  2448674  149658   VZWEWM_maxbin.066.fa   \n",
       "69330                 33.33  2452983   20640  VZXIHP_metabat.131.fa   \n",
       "69331                  0.00  1774424   55805   VZXIHP_metabat.45.fa   \n",
       "69332                 85.71  1595826    6604   VZXIHP_metabat.71.fa   \n",
       "\n",
       "         cluster           rep_MAG_name  \n",
       "69328     HQ-9_1   VZQLWU_vamb.11899.fa  \n",
       "69329    HQ-52_1   VZWEWM_maxbin.066.fa  \n",
       "69330  HQ-1099_1  VZXIHP_metabat.131.fa  \n",
       "69331  HQ-1187_1   VZXIHP_metabat.45.fa  \n",
       "69332  HQ-1822_1   VZXIHP_metabat.71.fa  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 2. Add info about representative genomes\n",
    "representative = pd.read_csv('input/2023_Pantiukh_EstMB_MAGsDB_HQcl.csv')\n",
    "# define cluster name for each representative MAGs \n",
    "mrg = pd.merge(representative, mrg_st1a, left_on='rep_MAG_name', right_on='bin_name', how='inner')\n",
    "mrg = mrg[['rep_MAG_name', 'cluster']]\n",
    "# add ino about rep_MAG to mrg1\n",
    "mrg_st2 = pd.merge(mrg_st1a, mrg, on='cluster', how='outer')\n",
    "\n",
    "mrg_st2.tail()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69328</th>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>pre10-73_1</td>\n",
       "      <td>93.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2279068</td>\n",
       "      <td>46770</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>HQ-9_1</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema_D</td>\n",
       "      <td>Treponema_D porcinum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69329</th>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>pre10-735_1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2448674</td>\n",
       "      <td>149658</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>HQ-52_1</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Oxalobacter</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69330</th>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>pre10-305_1</td>\n",
       "      <td>92.24</td>\n",
       "      <td>2.59</td>\n",
       "      <td>33.33</td>\n",
       "      <td>2452983</td>\n",
       "      <td>20640</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>HQ-1099_1</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Desulfobacterota</td>\n",
       "      <td>Desulfovibrionia</td>\n",
       "      <td>Desulfovibrionales</td>\n",
       "      <td>Desulfovibrionaceae</td>\n",
       "      <td>Bilophila</td>\n",
       "      <td>undefined</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69331</th>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>pre10-338_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1774424</td>\n",
       "      <td>55805</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>HQ-1187_1</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>Angelakisella</td>\n",
       "      <td>Angelakisella sp004554485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69332</th>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>pre10-454_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>6.35</td>\n",
       "      <td>85.71</td>\n",
       "      <td>1595826</td>\n",
       "      <td>6604</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>HQ-1822_1</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>Borkfalkiaceae</td>\n",
       "      <td>Gallimonas</td>\n",
       "      <td>Gallimonas faecium</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  pre_cluster  completeness  contamination  \\\n",
       "69328   VZQLWU_vamb.11899.fa   pre10-73_1         93.97           0.00   \n",
       "69329   VZWEWM_maxbin.066.fa  pre10-735_1        100.00           1.72   \n",
       "69330  VZXIHP_metabat.131.fa  pre10-305_1         92.24           2.59   \n",
       "69331   VZXIHP_metabat.45.fa  pre10-338_1         98.28           1.72   \n",
       "69332   VZXIHP_metabat.71.fa  pre10-454_1         91.38           6.35   \n",
       "\n",
       "       strain_heterogeneity   length     N50     bst_preCl_bin_name  \\\n",
       "69328                  0.00  2279068   46770   VZQLWU_vamb.11899.fa   \n",
       "69329                  0.00  2448674  149658   VZWEWM_maxbin.066.fa   \n",
       "69330                 33.33  2452983   20640  VZXIHP_metabat.131.fa   \n",
       "69331                  0.00  1774424   55805   VZXIHP_metabat.45.fa   \n",
       "69332                 85.71  1595826    6604   VZXIHP_metabat.71.fa   \n",
       "\n",
       "         cluster           rep_MAG_name    domain            phylum  \\\n",
       "69328     HQ-9_1   VZQLWU_vamb.11899.fa  Bacteria     Spirochaetota   \n",
       "69329    HQ-52_1   VZWEWM_maxbin.066.fa  Bacteria    Pseudomonadota   \n",
       "69330  HQ-1099_1  VZXIHP_metabat.131.fa  Bacteria  Desulfobacterota   \n",
       "69331  HQ-1187_1   VZXIHP_metabat.45.fa  Bacteria       Bacillota_A   \n",
       "69332  HQ-1822_1   VZXIHP_metabat.71.fa  Bacteria       Bacillota_A   \n",
       "\n",
       "                     class               order               family  \\\n",
       "69328         Spirochaetia      Treponematales      Treponemataceae   \n",
       "69329  Gammaproteobacteria     Burkholderiales     Burkholderiaceae   \n",
       "69330     Desulfovibrionia  Desulfovibrionales  Desulfovibrionaceae   \n",
       "69331           Clostridia     Oscillospirales      Ruminococcaceae   \n",
       "69332           Clostridia  Christensenellales       Borkfalkiaceae   \n",
       "\n",
       "               genus                         sp  \n",
       "69328    Treponema_D       Treponema_D porcinum  \n",
       "69329    Oxalobacter                  undefined  \n",
       "69330      Bilophila                  undefined  \n",
       "69331  Angelakisella  Angelakisella sp004554485  \n",
       "69332     Gallimonas         Gallimonas faecium  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 3. Add info about taxonomy\n",
    "\n",
    "# READ bacteria taxonomy\n",
    "gt = pd.read_csv('input/HQ_gtdbtk.bac120.summary.tsv', sep='\\t')\n",
    "gt = gt[['user_genome','classification','fastani_ani','closest_placement_reference','closest_placement_ani']]\n",
    "\n",
    "gt['domain'] = gt['classification'].str.split(';', expand=True)[0].str.replace(r'd__', '')\n",
    "gt['phylum'] = gt['classification'].str.split(';', expand=True)[1].str.replace(r'p__', '')\n",
    "gt['class'] = gt['classification'].str.split(';', expand=True)[2].str.replace(r'c__', '')\n",
    "gt['order'] = gt['classification'].str.split(';', expand=True)[3].str.replace(r'o__', '')\n",
    "gt['family'] = gt['classification'].str.split(';', expand=True)[4].str.replace(r'f__', '')\n",
    "gt['genus'] = gt['classification'].str.split(';', expand=True)[5].str.replace(r'g__', '')\n",
    "gt['sp'] = gt['classification'].str.split(';', expand=True)[6].str.replace(r's__', '')\n",
    "gt = gt.drop('classification', axis=1)\n",
    "\n",
    "# rename column genome\n",
    "gt = gt.rename(columns={'user_genome': 'rep_MAG_name'})\n",
    "# skip columns \n",
    "gt = gt[['rep_MAG_name', 'domain', 'phylum', 'class', 'order', 'family', 'genus', 'sp']]\n",
    "# fill NaN empty values with value\n",
    "gt['sp'] = gt['sp'].replace('', 'undefined')\n",
    "gt['genus'] = gt['genus'].replace('', 'undefined')\n",
    "gt['family'] = gt['family'].replace('', 'undefined')\n",
    "\n",
    "# add fa to rep_MAG_name\n",
    "gt['rep_MAG_name'] = gt['rep_MAG_name'].astype(str)+'.fa'\n",
    "\n",
    "# add ino about taxonomy to mrg_st2\n",
    "mrg_st3 = pd.merge(mrg_st2, gt, on='rep_MAG_name', how='outer')\n",
    "\n",
    "mrg_st3.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bin_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69328</th>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>pre10-73_1</td>\n",
       "      <td>93.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2279068</td>\n",
       "      <td>46770</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>HQ-9_1</td>\n",
       "      <td>VZQLWU_vamb.11899.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Spirochaetota</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Treponematales</td>\n",
       "      <td>Treponemataceae</td>\n",
       "      <td>Treponema_D</td>\n",
       "      <td>Treponema_D porcinum</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69329</th>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>pre10-735_1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2448674</td>\n",
       "      <td>149658</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>HQ-52_1</td>\n",
       "      <td>VZWEWM_maxbin.066.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Oxalobacter</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69330</th>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>pre10-305_1</td>\n",
       "      <td>92.24</td>\n",
       "      <td>2.59</td>\n",
       "      <td>33.33</td>\n",
       "      <td>2452983</td>\n",
       "      <td>20640</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>HQ-1099_1</td>\n",
       "      <td>VZXIHP_metabat.131.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Desulfobacterota</td>\n",
       "      <td>Desulfovibrionia</td>\n",
       "      <td>Desulfovibrionales</td>\n",
       "      <td>Desulfovibrionaceae</td>\n",
       "      <td>Bilophila</td>\n",
       "      <td>undefined</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69331</th>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>pre10-338_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1774424</td>\n",
       "      <td>55805</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>HQ-1187_1</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>Angelakisella</td>\n",
       "      <td>Angelakisella sp004554485</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69332</th>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>pre10-454_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>6.35</td>\n",
       "      <td>85.71</td>\n",
       "      <td>1595826</td>\n",
       "      <td>6604</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>HQ-1822_1</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>Borkfalkiaceae</td>\n",
       "      <td>Gallimonas</td>\n",
       "      <td>Gallimonas faecium</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    bin_name  pre_cluster  completeness  contamination  \\\n",
       "69328   VZQLWU_vamb.11899.fa   pre10-73_1         93.97           0.00   \n",
       "69329   VZWEWM_maxbin.066.fa  pre10-735_1        100.00           1.72   \n",
       "69330  VZXIHP_metabat.131.fa  pre10-305_1         92.24           2.59   \n",
       "69331   VZXIHP_metabat.45.fa  pre10-338_1         98.28           1.72   \n",
       "69332   VZXIHP_metabat.71.fa  pre10-454_1         91.38           6.35   \n",
       "\n",
       "       strain_heterogeneity   length     N50     bst_preCl_bin_name  \\\n",
       "69328                  0.00  2279068   46770   VZQLWU_vamb.11899.fa   \n",
       "69329                  0.00  2448674  149658   VZWEWM_maxbin.066.fa   \n",
       "69330                 33.33  2452983   20640  VZXIHP_metabat.131.fa   \n",
       "69331                  0.00  1774424   55805   VZXIHP_metabat.45.fa   \n",
       "69332                 85.71  1595826    6604   VZXIHP_metabat.71.fa   \n",
       "\n",
       "         cluster           rep_MAG_name    domain            phylum  \\\n",
       "69328     HQ-9_1   VZQLWU_vamb.11899.fa  Bacteria     Spirochaetota   \n",
       "69329    HQ-52_1   VZWEWM_maxbin.066.fa  Bacteria    Pseudomonadota   \n",
       "69330  HQ-1099_1  VZXIHP_metabat.131.fa  Bacteria  Desulfobacterota   \n",
       "69331  HQ-1187_1   VZXIHP_metabat.45.fa  Bacteria       Bacillota_A   \n",
       "69332  HQ-1822_1   VZXIHP_metabat.71.fa  Bacteria       Bacillota_A   \n",
       "\n",
       "                     class               order               family  \\\n",
       "69328         Spirochaetia      Treponematales      Treponemataceae   \n",
       "69329  Gammaproteobacteria     Burkholderiales     Burkholderiaceae   \n",
       "69330     Desulfovibrionia  Desulfovibrionales  Desulfovibrionaceae   \n",
       "69331           Clostridia     Oscillospirales      Ruminococcaceae   \n",
       "69332           Clostridia  Christensenellales       Borkfalkiaceae   \n",
       "\n",
       "               genus                         sp  bin_number_in_cluster  \n",
       "69328    Treponema_D       Treponema_D porcinum                      1  \n",
       "69329    Oxalobacter                  undefined                      1  \n",
       "69330      Bilophila                  undefined                      1  \n",
       "69331  Angelakisella  Angelakisella sp004554485                      1  \n",
       "69332     Gallimonas         Gallimonas faecium                      1  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 4. Add info about bins number in cluster\n",
    "bn = mrg_st3[['bin_name','cluster']].groupby('cluster').count()\n",
    "# rename column\n",
    "bn = bn.rename(columns={'bin_name': 'bin_number_in_cluster'})\n",
    "# merge it with mrg_st3\n",
    "mrg_st4 = pd.merge(mrg_st3, bn, on='cluster', how='outer')\n",
    "\n",
    "mrg_st4.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4a. Add info about bins quality\n",
    "conditions = [(mrg_st4['completeness'] > 90) & (mrg_st4['contamination'] < 5),\n",
    "        (mrg_st4['completeness'] <= 90) & (mrg_st4['contamination'] < 10) | \\\n",
    "        (mrg_st4['completeness'] > 90) & (mrg_st4['contamination'] >= 5) & (mrg_st4['contamination'] < 10),\n",
    "        (mrg_st4['contamination'] >= 10)]\n",
    "\n",
    "values = ['HQ', 'MQ', 'LQ']\n",
    "mrg_st4['bin_qual'] = np.select(conditions, values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SAVE FINAL TABLE with all bins went to LQcl\n",
    "mrg_st4.to_excel('results/HQcl_all_bins_all_info.xlsx', index=False)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Move to representatives MAGs from cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "      <th>bin_qual</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2252</th>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>pre10-338_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1774424</td>\n",
       "      <td>55805</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>HQ-1187_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>Angelakisella</td>\n",
       "      <td>Angelakisella sp004554485</td>\n",
       "      <td>1</td>\n",
       "      <td>HQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2253</th>\n",
       "      <td>VZXIHP_metabat.47.fa</td>\n",
       "      <td>pre10-749_1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1681331</td>\n",
       "      <td>16994</td>\n",
       "      <td>VZXIHP_metabat.47.fa</td>\n",
       "      <td>HQ-21_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>RF32</td>\n",
       "      <td>CAG-239</td>\n",
       "      <td>MGBC133411</td>\n",
       "      <td>MGBC133411 sp017935485</td>\n",
       "      <td>10</td>\n",
       "      <td>HQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254</th>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>pre10-454_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>6.35</td>\n",
       "      <td>85.71</td>\n",
       "      <td>1595826</td>\n",
       "      <td>6604</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>HQ-1822_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>Borkfalkiaceae</td>\n",
       "      <td>Gallimonas</td>\n",
       "      <td>Gallimonas faecium</td>\n",
       "      <td>1</td>\n",
       "      <td>MQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2255</th>\n",
       "      <td>VZXIHP_vamb.13313.fa</td>\n",
       "      <td>pre10-581_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1802309</td>\n",
       "      <td>62695</td>\n",
       "      <td>VZXIHP_vamb.13313.fa</td>\n",
       "      <td>HQ-541_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae_A</td>\n",
       "      <td>Aphodousia</td>\n",
       "      <td>Aphodousia sp905201055</td>\n",
       "      <td>3</td>\n",
       "      <td>HQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2256</th>\n",
       "      <td>VZXIHP_vamb.28684.fa</td>\n",
       "      <td>pre10-243_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2339340</td>\n",
       "      <td>167392</td>\n",
       "      <td>VZXIHP_vamb.28684.fa</td>\n",
       "      <td>HQ-1830_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>HGM11417</td>\n",
       "      <td>HGM11417</td>\n",
       "      <td>undefined</td>\n",
       "      <td>3</td>\n",
       "      <td>HQ</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              rep_MAG_name  pre_cluster  completeness  contamination  \\\n",
       "2252  VZXIHP_metabat.45.fa  pre10-338_1         98.28           1.72   \n",
       "2253  VZXIHP_metabat.47.fa  pre10-749_1        100.00           0.00   \n",
       "2254  VZXIHP_metabat.71.fa  pre10-454_1         91.38           6.35   \n",
       "2255  VZXIHP_vamb.13313.fa  pre10-581_1         98.28           0.00   \n",
       "2256  VZXIHP_vamb.28684.fa  pre10-243_1         98.28           0.00   \n",
       "\n",
       "      strain_heterogeneity   length     N50    bst_preCl_bin_name    cluster  \\\n",
       "2252                  0.00  1774424   55805  VZXIHP_metabat.45.fa  HQ-1187_1   \n",
       "2253                  0.00  1681331   16994  VZXIHP_metabat.47.fa    HQ-21_1   \n",
       "2254                 85.71  1595826    6604  VZXIHP_metabat.71.fa  HQ-1822_1   \n",
       "2255                  0.00  1802309   62695  VZXIHP_vamb.13313.fa   HQ-541_1   \n",
       "2256                  0.00  2339340  167392  VZXIHP_vamb.28684.fa  HQ-1830_1   \n",
       "\n",
       "        domain          phylum                class               order  \\\n",
       "2252  Bacteria     Bacillota_A           Clostridia     Oscillospirales   \n",
       "2253  Bacteria  Pseudomonadota  Alphaproteobacteria                RF32   \n",
       "2254  Bacteria     Bacillota_A           Clostridia  Christensenellales   \n",
       "2255  Bacteria  Pseudomonadota  Gammaproteobacteria     Burkholderiales   \n",
       "2256  Bacteria     Bacillota_A           Clostridia  Christensenellales   \n",
       "\n",
       "                  family          genus                         sp  \\\n",
       "2252     Ruminococcaceae  Angelakisella  Angelakisella sp004554485   \n",
       "2253             CAG-239     MGBC133411     MGBC133411 sp017935485   \n",
       "2254      Borkfalkiaceae     Gallimonas         Gallimonas faecium   \n",
       "2255  Burkholderiaceae_A     Aphodousia     Aphodousia sp905201055   \n",
       "2256            HGM11417       HGM11417                  undefined   \n",
       "\n",
       "      bin_number_in_cluster bin_qual  \n",
       "2252                      1       HQ  \n",
       "2253                     10       HQ  \n",
       "2254                      1       MQ  \n",
       "2255                      3       HQ  \n",
       "2256                      3       HQ  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 5. Select only representative MAGs\n",
    "mrg_st5 = pd.merge(representative, mrg_st4, left_on='rep_MAG_name', right_on='bin_name', how='inner')\n",
    "# rename column genome\n",
    "mrg_st5 = mrg_st5.rename(columns={'rep_MAG_name_x': 'rep_MAG_name'})\n",
    "# drop column centrality\n",
    "mrg_st5 = mrg_st5.drop(columns=['bin_name','rep_MAG_name_y'])\n",
    "mrg_st5.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep_MAG_name</th>\n",
       "      <th>pre_cluster</th>\n",
       "      <th>completeness</th>\n",
       "      <th>contamination</th>\n",
       "      <th>strain_heterogeneity</th>\n",
       "      <th>length</th>\n",
       "      <th>N50</th>\n",
       "      <th>bst_preCl_bin_name</th>\n",
       "      <th>cluster</th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>bin_number_in_cluster</th>\n",
       "      <th>bin_qual</th>\n",
       "      <th>rep_MAG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2252</th>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>pre10-338_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>1.72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1774424</td>\n",
       "      <td>55805</td>\n",
       "      <td>VZXIHP_metabat.45.fa</td>\n",
       "      <td>HQ-1187_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Oscillospirales</td>\n",
       "      <td>Ruminococcaceae</td>\n",
       "      <td>Angelakisella</td>\n",
       "      <td>Angelakisella sp004554485</td>\n",
       "      <td>1</td>\n",
       "      <td>HQ</td>\n",
       "      <td>H1581_Angelakisella_sp004554485.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2253</th>\n",
       "      <td>VZXIHP_metabat.47.fa</td>\n",
       "      <td>pre10-749_1</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1681331</td>\n",
       "      <td>16994</td>\n",
       "      <td>VZXIHP_metabat.47.fa</td>\n",
       "      <td>HQ-21_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>RF32</td>\n",
       "      <td>CAG-239</td>\n",
       "      <td>MGBC133411</td>\n",
       "      <td>MGBC133411 sp017935485</td>\n",
       "      <td>10</td>\n",
       "      <td>HQ</td>\n",
       "      <td>H0712_MGBC133411_sp017935485.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2254</th>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>pre10-454_1</td>\n",
       "      <td>91.38</td>\n",
       "      <td>6.35</td>\n",
       "      <td>85.71</td>\n",
       "      <td>1595826</td>\n",
       "      <td>6604</td>\n",
       "      <td>VZXIHP_metabat.71.fa</td>\n",
       "      <td>HQ-1822_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>Borkfalkiaceae</td>\n",
       "      <td>Gallimonas</td>\n",
       "      <td>Gallimonas faecium</td>\n",
       "      <td>1</td>\n",
       "      <td>MQ</td>\n",
       "      <td>H1845_Gallimonas_faecium.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2255</th>\n",
       "      <td>VZXIHP_vamb.13313.fa</td>\n",
       "      <td>pre10-581_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1802309</td>\n",
       "      <td>62695</td>\n",
       "      <td>VZXIHP_vamb.13313.fa</td>\n",
       "      <td>HQ-541_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiales</td>\n",
       "      <td>Burkholderiaceae_A</td>\n",
       "      <td>Aphodousia</td>\n",
       "      <td>Aphodousia sp905201055</td>\n",
       "      <td>3</td>\n",
       "      <td>HQ</td>\n",
       "      <td>H1204_Aphodousia_sp905201055.fa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2256</th>\n",
       "      <td>VZXIHP_vamb.28684.fa</td>\n",
       "      <td>pre10-243_1</td>\n",
       "      <td>98.28</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2339340</td>\n",
       "      <td>167392</td>\n",
       "      <td>VZXIHP_vamb.28684.fa</td>\n",
       "      <td>HQ-1830_1</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Christensenellales</td>\n",
       "      <td>HGM11417</td>\n",
       "      <td>HGM11417</td>\n",
       "      <td>undefined</td>\n",
       "      <td>3</td>\n",
       "      <td>HQ</td>\n",
       "      <td>H1164_HGM11417_undS.fa</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              rep_MAG_name  pre_cluster  completeness  contamination  \\\n",
       "2252  VZXIHP_metabat.45.fa  pre10-338_1         98.28           1.72   \n",
       "2253  VZXIHP_metabat.47.fa  pre10-749_1        100.00           0.00   \n",
       "2254  VZXIHP_metabat.71.fa  pre10-454_1         91.38           6.35   \n",
       "2255  VZXIHP_vamb.13313.fa  pre10-581_1         98.28           0.00   \n",
       "2256  VZXIHP_vamb.28684.fa  pre10-243_1         98.28           0.00   \n",
       "\n",
       "      strain_heterogeneity   length     N50    bst_preCl_bin_name    cluster  \\\n",
       "2252                  0.00  1774424   55805  VZXIHP_metabat.45.fa  HQ-1187_1   \n",
       "2253                  0.00  1681331   16994  VZXIHP_metabat.47.fa    HQ-21_1   \n",
       "2254                 85.71  1595826    6604  VZXIHP_metabat.71.fa  HQ-1822_1   \n",
       "2255                  0.00  1802309   62695  VZXIHP_vamb.13313.fa   HQ-541_1   \n",
       "2256                  0.00  2339340  167392  VZXIHP_vamb.28684.fa  HQ-1830_1   \n",
       "\n",
       "        domain          phylum                class               order  \\\n",
       "2252  Bacteria     Bacillota_A           Clostridia     Oscillospirales   \n",
       "2253  Bacteria  Pseudomonadota  Alphaproteobacteria                RF32   \n",
       "2254  Bacteria     Bacillota_A           Clostridia  Christensenellales   \n",
       "2255  Bacteria  Pseudomonadota  Gammaproteobacteria     Burkholderiales   \n",
       "2256  Bacteria     Bacillota_A           Clostridia  Christensenellales   \n",
       "\n",
       "                  family          genus                         sp  \\\n",
       "2252     Ruminococcaceae  Angelakisella  Angelakisella sp004554485   \n",
       "2253             CAG-239     MGBC133411     MGBC133411 sp017935485   \n",
       "2254      Borkfalkiaceae     Gallimonas         Gallimonas faecium   \n",
       "2255  Burkholderiaceae_A     Aphodousia     Aphodousia sp905201055   \n",
       "2256            HGM11417       HGM11417                  undefined   \n",
       "\n",
       "      bin_number_in_cluster bin_qual                          rep_MAG_ID  \n",
       "2252                      1       HQ  H1581_Angelakisella_sp004554485.fa  \n",
       "2253                     10       HQ     H0712_MGBC133411_sp017935485.fa  \n",
       "2254                      1       MQ         H1845_Gallimonas_faecium.fa  \n",
       "2255                      3       HQ     H1204_Aphodousia_sp905201055.fa  \n",
       "2256                      3       HQ              H1164_HGM11417_undS.fa  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Step 6. Add info about rep_MAG_ID\n",
    "mid = pd.read_csv('input/HQ_rename.csv', sep='\\t')\n",
    "mrg_st6 = pd.merge(mrg_st5, mid, on='rep_MAG_name', how='inner')\n",
    "mrg_st6.tail()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SAVE FINAL TABLE with only representative MAGs came from LQcl\n",
    "mrg_st6.to_excel('results/HQcl_representative_MAGs_all_info.xlsx', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
